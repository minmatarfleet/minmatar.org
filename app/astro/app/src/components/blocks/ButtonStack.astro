---
import Flexblock from "@components/compositions/Flexblock.astro";

---

<Flexblock class="[ button-stack ]" gap="var(--space-2xs)">
    <slot />
</Flexblock>

<style lang="scss" is:global>
    .button-stack {
        display: flex;
        width: fit-content;

        button,
        a {
            width: 100%;
        }
    }
</style>