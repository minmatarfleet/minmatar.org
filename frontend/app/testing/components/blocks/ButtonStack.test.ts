import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import ButtonStack from "@components/blocks/ButtonStack.astro";

test("ButtonStack defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(ButtonStack, {
    slots: {
      default: "chickens",
    },
  });

  expect(result).toMatchSnapshot();
});
