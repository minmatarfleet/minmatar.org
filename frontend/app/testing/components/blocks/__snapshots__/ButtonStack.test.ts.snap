// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonStack defaults 1`] = `"<div class="flex-block [ button-stack ] gap---space-2xs" data-mod style="--flex_block_gap: var(--space-2xs);">  chickens  </div>  "`;
